conseil-common/src/main/scala/tech/cryptonomic/conseil/common/bitcoin/BitcoinPersistence.scala

```diff
@@ -0,0 +1,140 @@
+      block: Block,
+      transactions: List[Transaction]
+  ): DBIOAction[Unit, NoStream, Effect.Write] =
+    DBIO.seq(
+      Tables.Blocks += block.convertTo[Tables.BlocksRow],
+      Tables.Transactions ++= transactions.map(_.convertTo[Tables.TransactionsRow]),
+      Tables.Inputs ++= transactions.flatMap(t => t.vin.map(c => (t, c))).map(_.convertTo[Tables.InputsRow]),
+      Tables.Outputs ++= transactions.flatMap(t => t.vout.map(c => (t, c))).map(_.convertTo[Tables.OutputsRow])
+    )
+}
+
+object BitcoinPersistence {
+
+  /**
+    * Create [[cats.Resource]] with [[BitcoinPersistence]].
+    */
+  def resource[F[_]: Concurrent]: Resource[F, BitcoinPersistence[F]] =
+    Resource.pure(new BitcoinPersistence[F])
+
+  /**
+    * Convert form [[Block]] to [[Tables.BlocksRow]]
+    * TODO: This conversion should be done with the Chimney,
+    *       but it's blocked due to the https://github.com/scala/bug/issues/11157
+    */
+  implicit val blockToBlocksRow: Conversion[Id, Block, Tables.BlocksRow] = new Conversion[Id, Block, Tables.BlocksRow] {
+    override def convert(from: Block) =
+      Tables.BlocksRow(
+        hash = from.hash,
+        size = from.size,
+        strippedSize = from.strippedsize,
+        weight = from.weight,
+        height = from.height,
+        version = from.version,
+        versionHex = from.versionHex,
+        merkleRoot = from.merkleroot,
+        nonce = from.nonce,
+        bits = from.bits,
+        difficulty = from.difficulty,
+        chainWork = from.chainwork,
+        nTx = from.nTx,
+        previousBlockHash = from.previousblockhash,
+        nextBlockHash = from.nextblockhash,
+        medianTime = new Timestamp(from.mediantime),
+        time = new Timestamp(from.time)
+      )
+  }
+
+  /**
+    * Convert form [[Transaction]] to [[Tables.TransactionsRow]]
+    * TODO: This conversion should be done with the Chimney,
+    *       but it's blocked due to the https://github.com/scala/bug/issues/11157
+    */
+  implicit val transactionToTransactionsRow: Conversion[Id, Transaction, Tables.TransactionsRow] =
+    new Conversion[Id, Transaction, Tables.TransactionsRow] {
+      override def convert(from: Transaction) =
+        Tables.TransactionsRow(
+          txid = from.txid,
+          blockhash = from.blockhash,
+          hash = from.hash,
+          hex = from.hex,
+          size = from.size,
+          vsize = from.vsize,
+          weight = from.weight,
+          version = from.version,
+          lockTime = new Timestamp(from.locktime),
+          blockTime = new Timestamp(from.blocktime),
+          time = new Timestamp(from.time)
+        )
+    }
+
+  /**
+    * Convert form [[TransactionInput]] to [[Tables.InputsRow]]
+    * TODO: This conversion should be done with the Chimney,
+    *       but it's blocked due to the https://github.com/scala/bug/issues/11157
+    */
+  implicit val inputToInputsRow: Conversion[Id, (Transaction, TransactionInput), Tables.InputsRow] =
```

Comment:
You can do more sophisticated things here, too

Like expecting to do a `Conversion[List, Map[TransactionId, TransactionInput], Tables.InputsRow]` from 
`Map[TransactionId, TransactionInput] => List[Tables.InputsRow]`

See if you like it